src/client/utils/fetcher.ts

```diff
@@ -14,6 +16,18 @@ export type IFetchComponent = {
   fetch(url: RequestInfo, init?: RequestOptions): Promise<Response>
 }
 
+export function withClientAgentInjection(fetcher: IFetchComponent): IFetchComponent {
+  return {
+    fetch: async (url: RequestInfo, init?: RequestOptions): Promise<Response> => {
+      const optionsWithInjectedClientAgent = mergeRequestOptions(init ? init : {}, {
+        headers: { 'Client-Agent': CURRENT_VERSION || 'Unknown' }
```

Comment:
```suggestion
        headers: { 'X-Requested-By': `dcl-catalyst-client/${CURRENT_VERSION || 'Unknown'}` }
```

Please use a common header (X-Requested-By is the convention). If not using the proposed, make sure to use a header starting with X- to prevent CORS blocks